pyogrio/raw.py

```diff
@@ -118,7 +119,7 @@ def read(
 
     if not "://" in path:
         if not "/vsi" in path.lower() and not os.path.exists(path):
-            raise ValueError(f"'{path}' does not exist")
+            raise DataSourceError(f"'{path}' does not exist")
```

Comment:
Changing this here to a DataSourceError for consistency, but, raising a builtin `FileNotFoundError` might actually be more appropriate ..

Reply:
Actually, just removing this code seems even the easier solution, as GDAL will handle this case fine and raise a consistent error that way. 
@brendan-ward do you remember why you added an explicit `os.path.exists(path)` check in the first place.

With the latest change removing the above check, we get a similar error message from GDAL:

```
In [2]: pyogrio.read_dataframe("non-existent.shp")
ERROR 4: non-existent.shp: No such file or directory
---------------------------------------------------------------------------
CPLE_OpenFailedError                      Traceback (most recent call last)
...
DataSourceError: non-existent.shp: No such file or directory
```